Config mgmt recipe (#894)
uyuni-project/uyuni-docs: modules/cookbook/pages/recipe-configmgmt.adoc

[[cookbook-config-mgmt]]
= Configuration Management

You can use configuration files and channels to manage configuration for your clients, rather than configuring each client manually.
This recipe shows you how to use the {productname} {webui} to create a centrally-managed configuration file, assign it to a Salt client, and apply the configuration.

== Problem

> If you are managing a lot of clients, you probably don't want to manually apply configuration settings to each of them in turn.

Reviewer: ```suggestion
If you are managing a lot of clients, you probably do not want to manually apply configuration settings to each of them in turn.
```

Configuration channels are used to organize configuration files.
You can subscribe clients to configuration channels, and deploy configuration files as required.

== Solution

When you have completed this recipe, you will have a configuration channel containing a configuration file, have assigned clients to the channel, and applied the configuration successfully.

== Before you begin

Before you start, you should already have:

* A {productname} Server installed, that you can access using the {webui}.
* At least one Salt client registered to your server.

> * Appropriate subscriptions from http://scc.suse.com for the products you are using.

Reviewer: I think this is optional (Uyuni).

Author: True, although there's a lot of caveats in that sentence ("appropriate" etc)

This recipe uses a centrally managed configuration file and a Salt state.
You can also use locally-managed configuration files,  traditional clients, and different methods, to get more flexibility in how you manage configuration in your environment.
For more information about the different ways to manage configuration, see xref:client-configuration:configuration-management.adoc[].

== Steps

Create a new configuration channel and file::

. In the {productname} {webui}, navigate to menu:Configuration[Channels] and click btn:[Create State Channel].

> . Type a name, label, and description for your configuration file, and type the contents of your configuration file.

Reviewer: I think you first want to enter the details of the channel, the cfg file will come later.

Author: Not really, it's a very weird UI.

> A simple example that you can copy is at the end of this section.

Reviewer: ```suggestion
An example that you can copy is at the end of this section.
```

. Click btn:[Create Config State Channel]
+
image::new_state.png[scaledwidth=80%]

Assign clients to the configuration channel::

. In the {productname} {webui}, navigate to menu:Systems[Systems List] and select the client you want to assign to your configuration channel.
. Navigate to the [guimenu]``Configuration`` tab.
  In the guimenu:``Configuration Overview`` section, click btn:[Subscribe to channels].
. Check the configuration channel you created earlier, and click btn:[Continue].
+
image::assign_client_cfg.png[scaledwidth=80%]

> . If you have more than one configuration channel, you can rank them in order of importance, or just click btn:[Update Channel Rankings] to finish.

Reviewer: ```suggestion
. If you have more than one configuration channel, you can rank them in order of importance, or click btn:[Update Channel Rankings] to finish.
```
